src/django/pfb_analysis/models.py

```diff
@@ -75,19 +82,62 @@ class Status(object):
             (ERROR, 'Error',),
         )
 
+    batch_job_id = models.CharField(max_length=256, blank=True, null=True)
+    neighborhood = models.ForeignKey(Neighborhood,
+                                     related_name='analysis_jobs',
+                                     on_delete=models.CASCADE)
     status = models.CharField(choices=Status.CHOICES,
                               default=Status.CREATED,
                               max_length=12,
                               help_text='The current status of the AnalysisJob')
 
-    neighborhood = models.ForeignKey(Neighborhood,
-                                     related_name='analysis_jobs',
-                                     on_delete=models.CASCADE)
-
     def run(self):
-        """ Run the analysis job
-
-        TODO: Implement
+        """ Run the analysis job, configuring ENV appropriately """
+        def create_environment(**kwargs):
+            return [{'name': k, 'value': v} for k, v in kwargs.iteritems()]
+
+        if self.batch_job_id is not None:
+            logger.warn('Attempt to re-run job: {}. Skipping.'.format(self.uuid))
+            return
+
+        client = boto3.client('batch')
+        job_definition = settings.PFB_AWS_BATCH_JOB_DEFINITION_NAME_REVISION
+        job_name = '{}:{}'.format(job_definition, str(self.uuid)[:8]).replace(':', '--')
```

Comment:
~~What about `str(self.uuid)[:8]` first, since that adds more information and would potentially cause more bother if it got totally cut off?~~

Just kidding, that would make them annoying to look for in CloudWatch.  What about `job_name = '{}--{}'.format(job_definition.replace(':', '--')[:40], str(self.uuid)[:8])` ?